test/TestHelper.cpp

```
@@ -547,26 +545,21 @@ bytes importData(json_spirit::mObject const& _o)
 
 std::string compileLLL(std::string const& _code)
 {
-	FILE *fp;
-	char path[1035];
-	FILE *ft = fopen("temp.sol", "w");
-	if (ft == NULL)
-		cerr << "Error creating temp file for lllc";
-	else
-		fputs(_code.c_str(), ft);
-	fclose(ft);
-
-	fp = popen("../../solidity/lllc/lllc temp.sol", "r");
+	char input[1024];
+	boost::filesystem::path path = boost::filesystem::unique_path();
+	std::string filename = path.string() + "/code.sol";
+	std::string lllc ("../../solidity/lllc/lllc ");
+	lllc += filename.c_str();
+	writeFile(filename, _code);
+
+	FILE *fp = popen(lllc.c_str(), "r");
 	if (fp == NULL)
-		cerr << "Failed to run lllc";
-
-	fgets(path, sizeof(path)-1, fp);
-
+		BOOST_ERROR("Failed to run lllc");
+	fgets(input, sizeof(input)-1, fp);
```

Please throw an exception if fgets returns null